src/BotPlutusInterface/Balance.hs

```diff
@@ -476,35 +496,24 @@ addOutput changeAddr tx =
     return $ tx {txOutputs = txOutputs tx ++ [changeTxOutWithMinAmt]}
 
 {- | Add the required signatories to the transaction. Be aware the the signature itself is invalid,
- and will be ignored. Only the pub key hashes are used, mapped to signing key files on disk.
+ and will be ignored. Only the pub key hashes are used, mapped to signing key files on disk if there's one.
+Otherwise, signing key used is a dummy one to support multi-sig schemes.
+WARN: This is brittle and should break in unexpected ways down the line.
 -}
```

Comment:
Why do we sign with dummy key instead of (a) throwing an error or (b) not doing anything?

Reply:
We need the key to be in the signatures, so we can sign properly later. Ideally we'd have a list, but we don't control the Tx type.
Throwing an error would be wrong, as we need to support partial signing

Reply:
OK, I get it